File: cloud-backend/src/apps/accounts/urls.rs
//! Route registration for the `accounts` app.

use djangors_core::Router;

use super::views;

/// Build the accounts router containing all authentication and token endpoints.
pub fn urls() -> Router {
    Router::new()
        .post("/auth/register", views::register)
        .post("/auth/login", views::login)
        .post("/auth/device", views::device_flow_init)
        .get("/auth/device/token", views::device_flow_poll)
        .post("/auth/device/authorize", views::device_flow_authorize)
        .post("/auth/token", views::create_api_token)
        .delete("/auth/token/{id}", views::revoke_api_token)
        .post("/auth/refresh", views::refresh_token)
        .get("/auth/me", views::me)
        .post("/auth/logout", views::logout)
}

File: examples/bloom_todo/apps/server/lib/apps/auth/urls.dart
import 'package:bloom_framework/bloom_server.dart';
import 'views.dart';

void registerUrls(
  BloomApiRouter router, {
  String prefix = '/api/auth',
}) {
  router.post('$prefix/signup', AuthViews.signup);
  router.post('$prefix/login', AuthViews.login);
  router.post('$prefix/refresh', AuthViews.refresh);
  router.post('$prefix/logout', AuthViews.logout);
  router.get('$prefix/me', AuthViews.me);
}
